src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -32,15 +32,17 @@ private LogFileGroup(List<LogFile> files, Pattern filePattern, Map<String, LogFi
 
     /**
      * Create a list of Logfiles that are sorted based on lastModified time.
-     * @param filePattern the fileNameRegex used for each component to recognize its log files.
-     * @param directoryURI the directory path of the log files of component.
-     * @param lastUpdated the saved updated time of the last uploaded log of a component.
-     * @param workDir component work directory
+     *
+     * @param componentLogConfiguration component log configuration
+     * @param lastUpdated               the saved updated time of the last uploaded log of a component.
+     * @param workDir                   component work directory
      * @return list of logFile.
      * @throws InvalidLogGroupException the exception if this is not a valid directory.
      */
-    public static LogFileGroup create(Pattern filePattern, URI directoryURI, Instant lastUpdated, Path workDir)
+    public static LogFileGroup create(ComponentLogConfiguration componentLogConfiguration,
```

Comment:
I guess this will allow us to use the component name to create unique folders right?

Reply:
That's right.